spyder/api/widgets/toolbars.py

```diff
@@ -129,6 +132,11 @@ class ApplicationToolBar(SpyderToolBar):
     """
     Spyder Main application ToolBar.
     """
+    ID = None
```

Comment:
Could we change this to `name` instead? It seems to me it corresponds to the toolbar name, right?

That way it'd be easier to remember this parameter, instead of `ID`.

Reply:
>Could we change this to name instead? It seems to me it corresponds to the toolbar name, right?

It corresponds to a unique identifier that is not exposed to the user and Qt uses to save and restore the state of the main window. I used the name ID to be in life with the `id()`  method in python. Even if Qt uses `setObjectName` as the method, the documentation explains that it needs to be unique, so uniqueness is the key property this attribute needs to provide.

<img width="981" alt="Screen Shot 2020-08-31 at 20 17 34" src="https://user-images.githubusercontent.com/3627835/91783862-039f1d00-ebc7-11ea-9aa4-256805fd5c08.png">


>That way it'd be easier to remember this parameter, instead of ID.

Added a docstring for the attribute to further explain what it does.

There is a check for the interface right below, and the error is explaining what a developer would need to do. If it was not set, or there was a typo an error would be raised explaining to the developer what needs to be changed.

This is going to be documented as part of the API.
```python
    def _check_interface(self):
        if self.ID is None:
            raise SpyderAPIError("Toolbar must define an ID attribute.")
```

If `ID` is not clear then `IDENTIFIER` is another option.

The projects API already uses this (and performs similar checks.
https://github.com/spyder-ide/spyder/blob/master/spyder/plugins/projects/api.py#L39

I also suggest the `NAME` attribute of plugin API be changed to `ID`, or `IDENTIFIER` since they provide a unique string. It is not really a name as that is provided by the `get_name` method.